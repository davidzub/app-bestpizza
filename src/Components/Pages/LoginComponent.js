import React, { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";
import InputComponent from "../General/InputComponent";
import Button from "react-bootstrap/Button";
import { runLogin } from "../../store/actions/login";
import logo from "../../images/Login-Best-Pizza.png";
import ic_constrasena from "../../images/ic_contrasena.png";
import ic_usuario from "../../images/ic_usuario.png";
import "./style.css";

const LoginComponent = () => {
  const dispatch = useDispatch();
  let history = useHistory();
  const validateUser = useSelector((state) => state.validateLogin);
  const [isLoading, setLoading] = useState(false);
  const [user, setUser] = useState();
  const [password, setPassword] = useState();
  useEffect(() => {
    if (validateUser) {
      history.push("/shops");
      if (isLoading) {
        setLoading(false);
      }
    } else {
      setLoading(false);
    }
  }, [validateUser, isLoading]);

  const handleClick = () => {
    dispatch(runLogin({ user, password }));
    setLoading(true);
  };
  return (
    <div id="loginForm">
      <div className="info">
        <img src={logo} className="logo" alt="" />
        <h3>Bienvenido</h3>
        <h4>A las mejores pizzas del país</h4>
      </div>
      <form noValidate autoComplete="off">
        <div className="field">
          <InputComponent
            type="email"
            placeholder="Usuario"
            icon={ic_usuario}
            disabled={isLoading}
            value={user}
            onChange={(value) => setUser(value)}
          />
        </div>
        <div className="field">
          <InputComponent
            type="password"
            placeholder="Contraseña"
            icon={ic_constrasena}
            disabled={isLoading}
            value={password}
            onChange={(value) => setPassword(value)}
          />
        </div>
        <p>¿Olvidaste tu contraseña?</p>
        <div>
          <Button
            variant="primary"
            disabled={isLoading}
            onClick={!isLoading ? handleClick : null}
          >
            {isLoading ? "Cargando..." : "Iniciar sesión"}
          </Button>
        </div>
      </form>
    </div>
  );
};
export default LoginComponent;
