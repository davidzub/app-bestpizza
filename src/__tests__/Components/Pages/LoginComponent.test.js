import React from "react";
import { mount } from "enzyme";
import configureStore from 'redux-mock-store'
import { Provider } from "react-redux";
import LoginComponent from "../../../Components/Pages/LoginComponent";
import thunk from "redux-thunk"

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
describe("<LoginComponent />", () => {
  let wrapper, store;
  const initialState = {
    validateLogin: false,
    listShops: null,
  };
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}><LoginComponent /></Provider>
    )
  });
  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Validate element 'loginForm'", () => {
    expect(wrapper.find("#loginForm")).toHaveLength(1);
  });
  test("click button", () => {
    const button = wrapper.find('button');
    expect(button.text()).toBe("Iniciar sesi√≥n");
    button.simulate('click');
    /* expect(button.prop('disabled')).toBeTruthy();  */
  });
});
