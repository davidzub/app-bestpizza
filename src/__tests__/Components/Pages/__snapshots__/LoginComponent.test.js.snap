// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginComponent /> render 1`] = `
<div
  id="loginForm"
>
  <div
    className="info"
  >
    <img
      alt=""
      className="logo"
      src="login-Best-Pizza.png"
    />
    <h3>
      Bienvenido
    </h3>
    <h4>
      A las mejores pizzas del país
    </h4>
  </div>
  <form
    autoComplete="off"
    noValidate={true}
  >
    <div
      className="field"
    >
      <input
        className="inputLogin"
        disabled={false}
        onChange={[Function]}
        placeholder="Usuario"
        style={
          Object {
            "backgroundImage": "url(ic_usuario.png)",
          }
        }
        type="email"
      />
    </div>
    <div
      className="field"
    >
      <input
        className="inputLogin"
        disabled={false}
        onChange={[Function]}
        placeholder="Contraseña"
        style={
          Object {
            "backgroundImage": "url(ic_contrasena.png)",
          }
        }
        type="password"
      />
    </div>
    <p>
      ¿Olvidaste tu contraseña?
    </p>
    <div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Iniciar sesión
      </button>
    </div>
  </form>
</div>
`;
